### PR TITLE
Implement translation for `PM_BEGIN_NODE`

### LINKED ISSUE #47: Implement translation for `PM_BEGIN_NODE`

### PR DESCRIPTION
**Pull request by egiurleo**
Closes #47 

### Motivation
Translates `PM_BEGIN_NODE`s from Prism to Sorbet.

These represent begin blocks, e.g.

```ruby
begin
  # ...
rescue
  # ...
end
```

Note that I haven't implemented support for `rescue` (#160), `ensure` (#86) or `else` because they're not needed for parsing the RBI gem.

### Test plan
See included automated tests.

### CODE REVIEW

File: test/prism_regression/begin.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# typed: true
+
+begin
+  foo
+# ^^^ error: Method `foo` does not exist on `T.class_of(<root>)`
```
Comment (amomchilov) [inline test/prism_regression/begin.rb:5]: You can use a string or integer literal, so you won't have to deal with this error

```suggestion
  1
  2
  3
```

File: test/prism_regression/begin.parse-tree.exp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Kwbegin {
+  stmts = [
+    Send {
+      receiver = NULL
+      method = <U foo>
+      args = [
+      ]
```
Comment (amomchilov) [inline test/prism_regression/begin.parse-tree.exp:7]:
```suggestion
      args = [
        Integer {
          val = "1"
        }
        Integer {
          val = "2"
        }
        Integer {
          val = "3"
        }
      ]
```

File: parser/prism/Translator.cc [hunk 3]
```diff
@@ -66,6 +66,23 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
...
+            NodeVec statements;
+
+            if (beginNode->statements != nullptr) {
+                auto body = beginNode->statements->body;
+                auto prismStmts = absl::MakeSpan(body.nodes, body.size);
+
+                for (auto &prismStmt : prismStmts) {
+                    statements.emplace_back(translate(prismStmt));
+                }
+            }
```
Comment (amomchilov) [inline parser/prism/Translator.cc:82]: I think you can reuse `translateStatements()` here, which handles other "begin-like" things like `()`, `#{ }`, method bodies, etc.:

```suggestion
            auto inlineIfSingle = false;
            auto statements = translateStatements(beginNode->statements, inlineIfSingle);
```

File: test/prism_regression/begin.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# typed: true
+
+begin
```
Comment (amomchilov) [inline test/prism_regression/begin.rb:3]: Could you also add a test case to ensure that empty bodies (to exercise the `node->body == nullptr` case) parse correctly, too?

```suggestion
begin
end

begin
```

**Review by amomchilov — APPROVED** [review]
Could you open an issue to track the translation of `else_clause `?

The `rescue` and `ensure` are their own node types, so we'll implement those in #160 and #86.

File: test/prism_regression/begin.parse-tree.exp (hunk 2 above)
Comment (egiurleo) [reply test/prism_regression/begin.parse-tree.exp:7]: Sorry what does this refer to?

**Comment by egiurleo** [comment]
@amomchilov Good point re: else. I actually [added a note to our rescue ticket](https://github.com/Shopify/sorbet/issues/160#issuecomment-2307644010) because the else clause works together with rescue.

File: test/prism_regression/begin.parse-tree.exp (hunk 2 above)
Comment (amomchilov) [reply test/prism_regression/begin.parse-tree.exp:7]: Just the test changes for https://github.com/Shopify/sorbet/pull/214#discussion_r1729291826
